src/components/TemplateLibrary/index.js

```diff
@@ -177,8 +177,37 @@ const LibraryComponent = () => {
     const template = await Template.fromUrl(templateIndex.ciceroUrl);
     const ciceroMark = templateToCiceroMark(template);
     const templateIdentifier = template.getIdentifier();
-    setup(ciceroMark, template);
-    saveTemplateToXml(templateIdentifier);
+    Office.context.document.customXmlParts.getByNamespaceAsync(CUSTOM_XML_NAMESPACE, result => {
+      if (result.status === Office.AsyncResultStatus.Succeeded) {
+        if (result.value.length === 0) {
+          setup(ciceroMark, template);
+          saveTemplateToXml(templateIdentifier);
+        }
+        else {
+          const customXmlPart = result.value[0];
+          customXmlPart.getNodesAsync('*/*', result => {
+            if (result.status === Office.AsyncResultStatus.Succeeded) {
+              let f=1;
+              if (result.value.length > 0) {
+                for (let node=0; node < result.value.length; ++node) {
+                  if (result.value[node].namespaceUri === templateIdentifier) {
+                    f=0;
+                  }
+                }
+              }
+              if(f){
+                setup(ciceroMark, template);
+                saveTemplateToXml(templateIdentifier);
+              }else{
+                console.info('Already present template');
+                // Office.context.ui.displayDialogAsync(`${window.location.origin}/bad-file.html`, { width: 30, height: 8 });
```

Comment:
I remember I did that initially. But I think it would be great if we can show toasts inside add-in. Just a suggestion. Let me know what you think.

Reply:
@algomaster99 
I searched how to implement toasts but couldn't find anything suitable.
As for toasts inside add-in, can we have a notifications bar where we show all the notifications like listeners attached and other necessary details?

Reply:
You can implement toasts inside add-in.

Reply:
@algomaster99 
I didn't get this part. Is there any reference which I can look up to?

Reply:
https://toasted-notes.netlify.app/

You can implement toasts like here on this link.

Reply:
![Screenshot (51)](https://user-images.githubusercontent.com/59891164/114564805-4c0ad700-9c8e-11eb-8006-917e88abbb0b.png)
@algomaster99 
Is this fine? The styling will be changed accordingly to make it look like error